cluster/kubernetes/sync.go

```diff
@@ -72,22 +363,45 @@ func (c *Kubectl) connectArgs() []string {
 	return args
 }
 
+// getDefaultNamespace returns the fallback namespace used by the
+// applied when a namespaced resource doesn't have one specified. This
+// is used when syncing to anticipate the identity of a resource in
+// the cluster given the manifest from a file (which may be missing
+// the namespace).
+func (k *Kubectl) getDefaultNamespace() string {
+	cmd := k.kubectlCommand("config", "get-contexts", "--no-headers")
+	out, err := cmd.Output()
+	if err != nil {
+		return DefaultDefaultNamespace
```

Comment:
I think we should let this function fail since
* `kubectl` is essential to `fluxd`
*  applying changes in `default` if there is a configured default namespace seems like a bad choice

Reply:
Yes, it should probably fail if `kubectl` can't be invoked -- good point.